### PR TITLE
WIP intermediate dev exchange

### PR DESCRIPTION
**Pull request by mpv92**

### CODE REVIEW

File: src/Command/VarnishControllerDeployCommand.php [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+            case 'false':
+                return false;
+            default:
+                return (bool) $value;
+        }
+    }
+
+    private function readFile(string $fileName): ?string
+    {
+        $content = @file_get_contents($fileName);
```
Comment (dbu) [inline src/Command/VarnishControllerDeployCommand.php:153]: i would prefer an exception over silently ignoring problems with reading the file

File: src/Command/VarnishControllerDeployCommand.php [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+            $id = $client->rollback($vclID, $rollbackID);
+            $logger->log('info', 'Rolled VCL {vclID} back to {id}', ['vclID' => $vclID, 'id' => $id]);
+
+            return 1;
+        }
+
+        $logger->log('info', 'Successfully deployed VCL {id} to group {groupID}', ['id' => $vclID, 'groupID' => $groupID]);
+        $logger->log('info', 'Compilation data: {data}', ['data' => Json::encode($compilationData)]);
+        $logger->log('info', 'Deployment data: {data}', ['data' => Json::encode($deployData)]);
```
Comment (dbu) [inline src/Command/VarnishControllerDeployCommand.php]: i would only print the data with `debug` verbosity, if at all.

why do we have two different "data"? can we find better names to make this more readable?

**Review by dbu — COMMENTED** [review]
i guess this is about the new varnish controller? looks good :+1: 

what is the state, does it work or is there work to do?

can you please add a description to this merge request that tells what this is about and if there are any steps left to do?

File: src/Command/VarnishControllerDeployCommand.php (hunk 2 above)
Comment (mpv92) [reply src/Command/VarnishControllerDeployCommand.php]: Hi David, thanks for the feedback, this and the @file_get_contents was 1:1 copied from the class before, i agree there's stuff that could be improved, but i rather focused on the API stuff. As i created a new File, it's more difficult to see the differences, i could also pack the changes in the already existing "VacClient.php" so it's easier to see the changes that were made?

Also this is an intermediate state for Emanuele, there's still the thing with "head" and "rollback" ongoing, which i am not sure how to solve.

File: src/Command/VarnishControllerDeployCommand.php [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+        $newVclSource = $this->readFile($vclInput);
+        if (!$newVclSource) {
+            throw new \InvalidArgumentException(sprintf('Filename "%s" is not readable.', $vclInput));
+        }
+        if ($newVclSource === $vclFile->getSource()) {
+            $logger->log('info', 'VCL did not change, will not update');
+        } else {
+            //todo
+            $rollbackID = $client->getHead($vclFile->getId());
```
Comment (mpv92) [inline src/Command/VarnishControllerDeployCommand.php]: @dbu this is still not adapted correctly

File: src/Command/VarnishControllerDeployCommand.php [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+        $logger->log(
+            'info', 'Deploying VCL {id} to group {groupID}', ['id' => $vclID, 'groupID' => $groupID]);
+        [$success, $compilationData, $deployData] = $client->deploy($groupID, $vclID);
+        if (!$success) {
+            $logger->log('error', 'Deployment failed');
+            $logger->log('error', 'Compilation errors: {errors}', ['errors' => Json::encode($compilationData)]);
+            $logger->log('error', 'Deployment errors: {errors}', ['errors' => Json::encode($deployData)]);
+
+            $id = $client->rollback($vclID, $rollbackID);
```
Comment (mpv92) [inline src/Command/VarnishControllerDeployCommand.php]: @dbu hence this rollback part won't work neither

File: src/Command/VarnishControllerDeployCommand.php [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+        if (!$groupID) {
+            throw new InvalidOptionException($vclGroup.' needs to be a valid group name');
+        }
+
+        $this->logger->log(
+            'info', 'Deploying VCL {id} to group {groupID}', ['id' => $vclFile->getId(), 'groupID' => $groupID]);
+
+        try {
+            $this->client->deploy($groupID, $vclFile->getId());
+            $this->logger->log('info', 'Successfully deployed VCL {id} to group {groupID}', ['id' => $vclFile->getId(), 'groupID' => $groupID]);
```
Comment (dbu) [inline src/Command/VarnishControllerDeployCommand.php:87]: why not use the `LoggerInterface::info` method anymore? i fnd that easier to read then passing the verbosity as string.

File: src/Command/VarnishControllerDeployCommand.php (hunk 2 above)
Comment (dbu) [reply src/Command/VarnishControllerDeployCommand.php]: i think keeping the data at debug verbosity could still be useful.

it depends on the overlap vs differences to know if its better to share code or have 2 separate things. if its 80% the same, it is easier to maintain if we share code. if its rather like 50% different, i find it better to have the 2 implementations not share code - or extract common functionality into a helper used by both.

**Comment by dbu** [comment]
closing in favor of #18

**Pull request closed without merging**